### PR TITLE
Add show-hidden-metrics-for-version to kubelet

### PR DESCRIPTION
**Pull request by serathius**
/kind feature
Ref https://github.com/kubernetes/kubernetes/issues/85270

**Does this PR introduce a user-facing change?:**
```release-note
New flag `--show-hidden-metrics-for-version` in kubelet can be used to show all hidden metrics that deprecated in the previous minor release.
```
**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:**
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/20190404-kubernetes-control-plane-metrics-stability.md
```
/cc @RainbowMango

### CODE REVIEW

**Review by mattjmcnaughton — COMMENTED** [review]
/unacc

Not sure I have the context to provide the requested review :)

At a high level, this implementation seems to be inline with the plan discussed in the KEP [here](https://github.com/kubernetes/enhancements/pull/1358). Wondering if the instrumentation folks who commented on the KEP should be the ones to give the sign-off?

cc @logicalhan @brancz

**Comment by mattjmcnaughton** [comment]
/test pull-kubernetes-node-e2e-containerd

Also, please rebase this diff when you get the chance :) thanks!

**Comment by brancz** [comment]
Looks good from instrumentation.

/lgtm

**Comment by RainbowMango** [comment]
/hold
We need to solve [the issue](https://github.com/kubernetes/kubernetes/issues/85411) first.

**Comment by RainbowMango** [comment]
@serathius 
Please rebase and solve the conflicts. :)

/hold cancel
/milestone v1.18
/priority important-soon
/assign @yujuhong @dashpole

**Comment by dashpole** [comment]
You will want to add this to the kubelet component configuration as well:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/config/types.go#L74

You can probably follow this change as an example:
https://github.com/kubernetes/kubernetes/commit/89dfd244838f21ecaf9bc9bf28662a7cccdd8425#diff-5088ffe9a3539b287e74e2ca03c5090c

**Comment by serathius** [comment]
@dashpole Addressed your commend. PTAL

**Comment by dashpole** [comment]
This lgtm after the tests are fixed.

**Comment by RainbowMango** [comment]
It seems this ready to go now.
@dashpole Can you take a look again.

**Comment by serathius** [comment]
Ping @erictune @yujuhong

**Review by erictune — APPROVED** [review]

File: staging/src/k8s.io/kubelet/config/v1beta1/types.go
Comment (lavalamp) [inline staging/src/k8s.io/kubelet/config/v1beta1/types.go]: Please be consistent with the other fields, this should be:
```suggestion
	ShowHiddenMetricsForVersion string `json:"showHiddenMetricsForVersion,omitempty"`
```

File: api/api-rules/violation_exceptions.list
Comment (lavalamp) [inline api/api-rules/violation_exceptions.list]: I don't think there's a need to violate the naming rule here.

File: staging/src/k8s.io/kubelet/config/v1beta1/types.go
Comment (serathius) [reply staging/src/k8s.io/kubelet/config/v1beta1/types.go]: Done

File: api/api-rules/violation_exceptions.list
Comment (serathius) [reply api/api-rules/violation_exceptions.list]: Done

**Pull request merged**
